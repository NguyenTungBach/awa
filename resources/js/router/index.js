import Vue from 'vue';
import VueRouter from 'vue-router';

Vue.use(VueRouter);

import PageNotFound from './modules/PageNotFound';
import Login from './modules/Login';
import ShiftManagement from './modules/ShiftManagement';
import DataManagement from './modules/DataManagement';
import Dev from './modules/Dev';

export const constantRoutes = [
	Login,
	Dev,
];

export const asyncRoutes = [
	ShiftManagement,
	DataManagement,
	{
		path: '/',
		name: 'Dashboard',
		meta: {
			roles: ['admin', 'driver'],
		},
		redirect: { name: 'ListShift' },
		hidden: true,
	},
	PageNotFound,
	{
		path: '*',
		hidden: true,
		redirect: { name: 'PageNotFound' },
	},
];

const createRouter = () => new VueRouter({
	mode: 'history',
	scrollBehavior: () => ({ y: 0 }),
	routes: constantRoutes,
});

const router = createRouter();

export function resetRouter() {
	const newRouter = createRouter();
	router.matcher = newRouter.matcher;
}

export default router;
