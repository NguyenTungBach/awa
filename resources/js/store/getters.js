const getters = {
	loading: state => state.loading.overlay,
	paddingShift: state => state.loading.paddingShift,

	language: state => state.app.language,
	pickerYearMonth: state => state.app.pickerYearMonth,
	pickerWeek: state => state.app.pickerWeek,
	permissionRoutes: state => state.permissions.routes,
	addRoutes: state => state.permissions.addRoutes,
	token: state => state.login.token,
	profile: state => state.login.profile,

	tableListShift: state => state.listShift.isTable,
	weekOrMonthListShift: state => state.listShift.isWeekOrMonth,
	reloadTableListShift: state => state.listShift.reLoadTable,

	tabIndexDriver: state => state.listDriver.tabIndex,

	listUpdateShift: state => state.listShift.listUpdate,
	listUpdateDayoff: state => state.listDayoff.listUpdate,
	listUpdateSchedule: state => state.listSchedule.listUpdate,
	listUpdateShiftCourseBase: state => state.listShift.listUpdate,
	listCoursePattern: state => state.listCoursePattern.list,

	showModalResponseAI: state => state.handleAI.showModalMessage,
	typeMessage: state => state.handleAI.typeMessage,
	textMessge: state => state.handleAI.textMessage,

	clickBackCalendarMonth: state => state.calendar.clickBack,
	clickNextCalendarMonth: state => state.calendar.clickNext,
	indexCalendarWeekListShift: state => state.calendar.indexCalendarWeekListShift,

	warningNotSaveUser: state => state.user.warningNotSave,
	warningNotSaveDriver: state => state.listDriver.warningNotSave,
	warningNotSaveCourse: state => state.course.warningNotSave,
	warningNotSaveCoursePattern: state => state.listCoursePattern.warningNotSave,
	idRouter: state => state.listCash.idRouter,
};

export default getters;
