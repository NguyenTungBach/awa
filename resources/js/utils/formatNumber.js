function formatNumberCode(string, maxlength = 4) {
	const number = parseInt(string);

	const re = /^([0-9]\d*)$/;

	if (re.test(number)) {
		return String(number).padStart(maxlength, '0');
	}

	return number;
}

function formatNumber(number) {
	if (number) {
		number = BigInt(number);
		return String(number).replace(/(.)(?=(\d{3})+$)/g, '$1,');
	}

	return '0';
}

export {
	formatNumberCode,
	formatNumber
};
