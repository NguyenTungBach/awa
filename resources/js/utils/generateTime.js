import i18n from '@/lang';

export function format2Digit(val) {
	return val < 10 ? `0${val}` : `${val}`;
}

export function generateTimeSelect(step = 5) {
	const list = [
		{
			value: null,
			text: i18n.t('APP.PLEASE_SELECT'),
		},
	];

	const len = 24;
	let idx = 0;

	while (idx < len) {
		for (let min = 0; min < 60; min += step) {
			list.push({
				value: `${format2Digit(idx)}:${format2Digit(min)}`,
				text: `${format2Digit(idx)}:${format2Digit(min)}`,
			});
		}

		idx++;
	}

	return list;
}
