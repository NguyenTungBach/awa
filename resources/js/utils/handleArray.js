function removeDirtyItemInArray(arr, dirty = [null, undefined, ''], key = null) {
	const results = arr.filter(element => {
		return !dirty.includes(key ? element[key] : element);
	});

	return results;
}

export {
	removeDirtyItemInArray
};
