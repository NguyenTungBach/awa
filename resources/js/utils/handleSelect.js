function convertValueToText(arr, value, keyValue = 'value', keyText = 'text') {
	let idx = 0;
	const len = arr.length;

	while (idx < len) {
		if (arr[idx][keyValue] === value) {
			return arr[idx][keyText];
		}

		idx++;
	}

	return '';
}

function convertArrValueToArrText(arr, arrValue, keyValue = 'value', keyText = 'text') {
	let idx = 0;
	const len = arrValue.length;

	const result = [];

	while (idx < len) {
		const item = convertValueToText(arr, arrValue[idx], keyValue, keyText);

		console.log(arrValue[idx]);

		result.push(item);

		idx++;
	}

	return result;
}

export {
	convertValueToText,
	convertArrValueToArrText
};
