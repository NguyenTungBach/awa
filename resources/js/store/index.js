import Vue from 'vue';
import Vuex from 'vuex';

import app from './modules/app';
import loading from './modules/loading';
import login from './modules/login';
import permissions from './modules/permissions';
import listShift from './modules/listShift';
import listDayoff from './modules/listDayoff';
import listSchedule from './modules/listSchedule';
import listDriver from './modules/listDriver';
import listCoursePattern from './modules/listCoursePattern';
import handleAI from './modules/handleAI';
import calendar from './modules/calendar';
import user from './modules/user';
import course from './modules/course';
import listCash from './modules/listCashOut';

import getters from './getters';

Vue.use(Vuex);

const modules = {
	app,
	loading,
	login,
	permissions,
	listShift,
	listDayoff,
	listSchedule,
	listDriver,
	listCoursePattern,
	handleAI,
	calendar,
	user,
	course,
	listCash,
};

const store = new Vuex.Store({
	modules,
	getters,
});

export default store;
