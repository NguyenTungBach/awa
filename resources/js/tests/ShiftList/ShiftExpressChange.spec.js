import store from '@/store';
import router from '@/router';
import ListShift from '@/pages/ShiftManagement/ListShift/index';
import { mount, createLocalVue } from '@vue/test-utils';
import CONSTANT from '@/const';
import { convertValueToText } from '@/utils/handleSelect';

describe('TEST COMPONENT LIST SHIFT', () => {
	test('Check render page', () => {
		const localVue = createLocalVue();
		const handleGetListCalendar = jest.fn();
		const handleGetListShift = jest.fn();
		// const initData = jest.fn();
		const wrapper = mount(ListShift, {
			localVue,
			store,
			router,
			methods: {
				handleGetListCalendar,
				handleGetListShift,
				// initData,
			},
		});

		const PAGE = wrapper.find('.list-shift');
		expect(PAGE.exists()).toBe(true);

		wrapper.destroy();
	});

	test('Check render title page', () => {
		const localVue = createLocalVue();
		const handleGetListCalendar = jest.fn();
		const handleGetListShift = jest.fn();
		const wrapper = mount(ListShift, {
			localVue,
			store,
			router,
			methods: {
				handleGetListCalendar,
				handleGetListShift,
			},
			data() {
				return {
					selectTable: 'CONSTANT.LIST_SHIFT.HIGHT_WAY_FEE',
					// handleGetListCalendar: jest.fn(),
				};
			},
		});

		const TITLE = wrapper.find('.title-page');
		expect(TITLE.text()).toEqual('LIST_SHIFT.TITLE_LIST_SHIFT_PRACTICAL_RECORD_TABLE');

		wrapper.destroy();
	});

	// test('Check render button choose week or month', () => {
	// 	const localVue = createLocalVue();
	// 	const wrapper = mount(ListShift, {
	// 		localVue,
	// 		store,
	// 		router,
	// 	});
	//
	// 	const BUTTON_WEEK = wrapper.find('.btn-list-shift-week');
	// 	expect(BUTTON_WEEK.exists()).toBe(true);
	// 	expect(BUTTON_WEEK.text()).toEqual('LIST_SHIFT.BUTTON_MONTH');
	//
	// 	const BUTTON_MONTH = wrapper.find('.btn-list-shift-month');
	// 	expect(BUTTON_MONTH.exists()).toBe(true);
	// 	expect(BUTTON_MONTH.text()).toEqual('LIST_SHIFT.BUTTON_WEEK');
	//
	// 	wrapper.destroy();
	// });

	// test('Check click button choose week or month', async() => {
	// 	const onClickSelectWeekMonth = jest.fn();
	// 	const localVue = createLocalVue();
	// 	const wrapper = mount(ListShift, {
	// 		localVue,
	// 		store,
	// 		router,
	// 		methods: {
	// 			onClickSelectWeekMonth,
	// 		},
	// 	});
	//
	// 	const BUTTON_WEEK = wrapper.find('.btn-list-shift-week');
	// 	await BUTTON_WEEK.trigger('click');
	// 	expect(onClickSelectWeekMonth).toHaveBeenCalled();
	//
	// 	const BUTTON_MONTH = wrapper.find('.btn-list-shift-month');
	// 	await BUTTON_MONTH.trigger('click');
	// 	expect(onClickSelectWeekMonth).toHaveBeenCalled();
	//
	// 	wrapper.destroy();
	// });

	// test('Check render button AI', () => {
	// 	const localVue = createLocalVue();
	// 	const wrapper = mount(ListShift, {
	// 		localVue,
	// 		store,
	// 		router,
	// 	});
	//
	// 	const CONTROL = wrapper.find('.zone-right');
	// 	expect(CONTROL.exists()).toBe(true);
	// 	const BUTTON = CONTROL.find('.show-text');
	// 	expect(BUTTON.exists()).toBe(true);
	// 	const TEXT_BUTTON = BUTTON.find('span');
	// 	expect(TEXT_BUTTON.text()).toEqual('LIST_SHIFT.BUTTON_DOWNLOAD_EXCEL');
	//
	// 	wrapper.destroy();
	// });
	//
	// test('Check click button AI', async() => {
	// 	const onExportExcel = jest.fn();
	// 	const localVue = createLocalVue();
	// 	const wrapper = mount(ListShift, {
	// 		localVue,
	// 		store,
	// 		router,
	// 		methods: {
	// 			onExportExcel,
	// 		},
	// 	});
	//
	// 	// const BUTTON = wrapper.find('.fa-robot');
	// 	const CONTROL = wrapper.find('.zone-right');
	// 	expect(CONTROL.exists()).toBe(true);
	// 	const BUTTON = CONTROL.find('.show-text');
	// 	expect(BUTTON.exists()).toBe(true);
	// 	await BUTTON.trigger('click');
	// 	expect(onExportExcel).toHaveBeenCalled();
	//
	// 	wrapper.destroy();
	// });

	// test('Check render button Excel', () => {
	// 	const localVue = createLocalVue();
	// 	const wrapper = mount(ListShift, {
	// 		localVue,
	// 		store,
	// 		router,
	// 	});
	//
	// 	const BUTTON = wrapper.find('.btn-excel');
	// 	expect(BUTTON.exists()).toBe(true);
	// 	expect(BUTTON.text()).toEqual('LIST_SHIFT.BUTTON_DOWNLOAD_EXCEL');
	//
	// 	wrapper.destroy();
	// });

	test('Check click button Excel', async() => {
		const onExportExcel = jest.fn();
		const onClickSelectTable = jest.fn();
		const initData = jest.fn();
		const handleGetListCalendar = jest.fn();
		const handleGetListShift = jest.fn();
		const localVue = createLocalVue();
		const wrapper = mount(ListShift, {
			localVue,
			store,
			router,
			methods: {
				onExportExcel,
				initData,
				onClickSelectTable,
				handleGetListCalendar,
				handleGetListShift,
				// handleGetListCalendar,
				// handleGetListShift,
				// onClickSelectWeekMonth,
			},
			data() {
				return {
					selectTable: 'CONSTANT.LIST_SHIFT.SHIFT_TABLE',
					// handleGetListCalendar: jest.fn(),
				};
			},
			computed: {
				role() {
					return 'admin';
				},
			},
		});

		const BUTTON_LIST = wrapper.find('.list-shift__control');
		expect(BUTTON_LIST.exists()).toEqual(true);
		expect(wrapper).toMatchSnapshot();
		const BUTTON_EXPRESS_CHANGE = wrapper.find('div > div:nth-child(1) > div > div > button:nth-child(2)');
		expect(BUTTON_EXPRESS_CHANGE.exists()).toEqual(true);
		await BUTTON_EXPRESS_CHANGE.trigger('click');
		// expect(wrapper.vm.onClickSelectTable).toHaveBeenCalled();

		// const TITLE = wrapper.find('.title-page');
		// expect(TITLE.text()).toEqual('LIST_SHIFT.TITLE_LIST_SHIFT_PRACTICAL_RECORD_TABLE');

		// console.log('akjgkaj: ', wrapper.vm.selectTable);
		// await wrapper.setData( {selectTable: 'CONSTANT.LIST_SHIFT.HIGHT_WAY_FEE'} );
		// expect(wrapper.vm.selectTable).toBe('CONSTANT.LIST_SHIFT.HIGHT_WAY_FEE');

		expect(wrapper).toMatchSnapshot();

		const BUTTON_WEEK = wrapper.find('.btn-excel');
		await BUTTON_WEEK.trigger('click');
		expect(onExportExcel).toHaveBeenCalled();

		// await BUTTON.trigger('click');
		// expect(onExportExcel).toHaveBeenCalled();

		wrapper.destroy();
	});

	// test('Check render button PDF', () => {
	// 	const localVue = createLocalVue();
	// 	const wrapper = mount(ListShift, {
	// 		localVue,
	// 		store,
	// 		router,
	// 	});

	// 	const CONTROL = wrapper.find('.zone-right');
	// 	expect(CONTROL.exists()).toBe(true);
	// 	const BUTTON = CONTROL.find('.show-text');
	// 	expect(BUTTON.exists()).toBe(true);
	// 	const TEXT_BUTTON = BUTTON.find('span');
	// 	expect(TEXT_BUTTON.text()).toEqual('LIST_SHIFT.BUTTON_DOWNLOAD_EXCEL');

	// 	wrapper.destroy();
	// });

	// test('Check click button PDF', async() => {
	// 	const onExportPDF = jest.fn();
	// 	const localVue = createLocalVue();
	// 	const wrapper = mount(ListShift, {
	// 		localVue,
	// 		store,
	// 		router,
	// 		methods: {
	// 			onExportPDF,
	// 		},
	// 	});

	// 	// const BUTTON_WEEK = wrapper.find('.btn-pdf');
	// 	// await BUTTON_WEEK.trigger('click');
	// 	// expect(onExportPDF).toHaveBeenCalled();
	// 	const CONTROL = wrapper.find('.zone-right');
	// 	expect(CONTROL.exists()).toBe(true);
	// 	const BUTTON = CONTROL.find('.show-text');
	// 	expect(BUTTON.exists()).toBe(true);
	// 	await BUTTON.trigger('click');
	// 	expect(onExportPDF).toHaveBeenCalled();

	// 	wrapper.destroy();
	// });

	test('Check click button choose table', async() => {
		const onExportExcel = jest.fn();
		const handleGetListCalendar = jest.fn();
		const handleGetListShift = jest.fn();
		const localVue = createLocalVue();
		const wrapper = mount(ListShift, {
			localVue,
			store,
			router,
			methods: {
				onExportExcel,
				handleGetListCalendar,
				handleGetListShift,
			},
		});

		const BUTTON_WEEK = wrapper.find('.btn-excel');
		await BUTTON_WEEK.trigger('click');
		expect(onExportExcel).toHaveBeenCalled();

		wrapper.destroy();
	});

	// test('Check render table title date', () => {
	// 	const localVue = createLocalVue();
	// 	const wrapper = mount(ListShift, {
	// 		localVue,
	// 		store,
	// 		router,
	// 	});
	//
	// 	const TABLE = wrapper.find('.zone-table table');
	// 	expect(TABLE.exists()).toBe(true);
	//
	// 	const LIST_HEADER = TABLE.findAll('th');
	//
	// 	// const CONST_LIST_HEADER = [
	// 	// 	'LIST_SHIFT.TABLE_DATE_EMPLOYEE_NUMBER',
	// 	// 	'LIST_SHIFT.TABLE_FLAG',
	// 	// 	'LIST_SHIFT.TABLE_FULL_NAME',
	// 	// ];
	//
	// 	expect(LIST_HEADER.length).toBe(67);
	// 	// for (let i = 0; i < LIST_HEADER.length; i++) {
	// 	// 	expect(LIST_HEADER.at(i).text()).toEqual(CONST_LIST_HEADER[i]);
	// 	// }
	//
	// 	wrapper.destroy();
	// });

	test('Check render body table', () => {
		const DATA = [
			{
				'driver_code': '0004',
				'driver_id': 4,
				'driver_name': 'Associate company',
				'type': 4,
				'typeName': 'associate company',
				'dataShift': {
					'driver_id': 4,
					'data_by_date': [
						{
							'driver_id': 4,
							'date': '2023-08-01',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-02',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-03',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-04',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-05',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-06',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-07',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-08',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-09',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-10',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-11',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-12',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-13',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-14',
							'course_ids': '11',
							'course_names': 'Course name 4',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-15',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-16',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-17',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-18',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-19',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-20',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-21',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-22',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-23',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-24',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-25',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-26',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-27',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-28',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-29',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-30',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
						{
							'driver_id': 4,
							'date': '2023-08-31',
							'course_ids': '',
							'course_names': '',
							'course_names_color': '',
						},
					],
				},
				'total_money': '80.00',
			},
		];
		const localVue = createLocalVue();
		const wrapper = mount(ListShift, {
			localVue,
			store,
			router,
			data() {
				return {
					listShift: DATA,
					// handleGetListCalendar: jest.fn(),
					initData: jest.fn(),
				};
			},
		});

		// jest.spyOn(wrapper.vm, 'handleGetListCalendar');

		const ZONE_TABLE = wrapper.find('.zone-table');
		const TABLE = ZONE_TABLE.find('.shift-table');

		// Kiểm tra HEAD table
		const THEAD = TABLE.find('thead');
		expect(THEAD.exists()).toBe(true);

		const TR = THEAD.findAll('tr');
		expect(TR.length).toBe(2);

		const TR_TH = TR.at(1).findAll('th');
		expect(TR_TH.exists()).toBe(true);

		// Kiểm tra tên 3 cột
		expect(TR_TH.at(0).text()).toEqual('LIST_SHIFT.TABLE_DATE_EMPLOYEE_NUMBER');
		expect(TR_TH.at(1).text()).toEqual('LIST_SHIFT.TABLE_FLAG');
		expect(TR_TH.at(2).text()).toEqual('LIST_SHIFT.TABLE_FULL_NAME');

		// Kiểm tra tên Data đổ ra
		const ROWS = TABLE.findAll('tbody tr');
		expect(ROWS.length).toBe(1);

		const len2 = DATA.length;
		let idx2 = 0;

		// kiểm tra 3 cột render đầu tiên
		while (idx2 < len2) {
			const ROW = ROWS.at(idx2);

			const COLUMNS = ROW.findAll('td');

			const COLUMN_driver_code = COLUMNS.at(0);
			expect(COLUMN_driver_code.text()).toEqual(DATA[idx2].driver_code);
			const COLUMN_CUSTOMER_NAME = COLUMNS.at(1);
			expect(COLUMN_CUSTOMER_NAME.text()).toEqual(convertValueToText(CONSTANT.LIST_DRIVER.LIST_FLAG, DATA[idx2].type));
			const COLUMN_Driver_name = COLUMNS.at(2);
			expect(COLUMN_Driver_name.text()).toEqual(DATA[idx2].driver_name);
			idx2++;
		}

		wrapper.destroy();
	});
});
