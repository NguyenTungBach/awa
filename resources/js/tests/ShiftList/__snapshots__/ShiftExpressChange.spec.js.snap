// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST COMPONENT LIST SHIFT Check click button Excel 1`] = `
<div class="col">
  <div class="list-shift">
    <div class="list-shift__header">
      <div class="row">
        <div class="col-sm-12 col-md-8 col-lg-8 col-xl-8 col-12">
          <div class="zone-left">
            <div class="zone-title"><span class="title-page">
                                LIST_SHIFT.TITLE_LIST_SHIFT_PRACTICAL_RECORD_TABLE
                            </span></div>
            <div class="zone-select-week-month"></div>
            <div class="zone-calendar-week"></div>
          </div>
        </div>
        <div class="col-sm-12 col-md-4 col-lg-4 col-xl-4 col-12">
          <!---->
          <!---->
          <div class="zone-right" style="display: none;">
            <div class="item-function btn-excel">
              <div class="show-icon"><i class="fas fa-file-excel"></i></div>
              <div class="show-text"><span>LIST_SHIFT.BUTTON_DOWNLOAD_EXCEL</span></div>
            </div>
          </div>
          <!---->
        </div>
      </div>
    </div>
    <div class="component-line-gray"></div>
    <div class="list-shift__control">
      <div class="row">
        <div class="col">
          <div class="text-left">
            <div role="group" class="button-text-left btn-group"><button type="button" class="btn btn-secondary control-button-group"><span>LIST_SHIFT.BUTTON_SHIFT_TABLE</span></button> <button type="button" class="btn btn-secondary control-button-group"><span>LIST_SHIFT.BUTTON_HIGHT_WAY_FEE</span></button> <button type="button" class="btn btn-secondary control-button-group"><span>LIST_SHIFT.BUTTON_TABLE_SALES</span></button> <button type="button" class="btn btn-secondary control-button-group"><span>LIST_SHIFT.BUTTON_PAYMENT</span></button></div>
          </div>
        </div>
        <div class="col">
          <div class="text-right"><button type="button" class="btn btn-edit btn-color-active btn-secondary rounded-pill" style="display: none;">
              LIST_SHIFT.BUTTON_EDIT
            </button></div>
          <!---->
        </div>
      </div>
    </div>
    <div class="list-shift__table">
      <div class="b-overlay-wrap position-relative">
        <div class="zone-table">
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
        <transition-stub name="" appear="true" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade" appearclass="" appearactiveclass="fade" appeartoclass="show">
          <!---->
        </transition-stub>
      </div>
    </div>
  </div>
  <div id="modal-closing-date___BV_modal_outer_" style="position: absolute; z-index: 1040;">
    <transition-stub enterclass="" leaveclass="" entertoclass="" leavetoclass="" enteractiveclass="" leaveactiveclass="">
      <div id="modal-closing-date" role="dialog" aria-hidden="true" aria-describedby="modal-closing-date___BV_modal_body_" class="modal fade" style="display: none;">
        <div class="modal-dialog modal-md">
          <!---->
          <div id="modal-closing-date___BV_modal_content_" tabindex="-1" class="modal-content">
            <!---->
            <div id="modal-closing-date___BV_modal_body_" class="modal-body modal-closing-date">
              <div class="text-center">
                <h5 class="font-weight-bolde">
                  締日を選択してください
                </h5>
              </div>
              <div class="body-item">
                <div class="row">
                  <div class="col-8" style="margin: auto;">
                    <div role="group" class="input-group">
                      <!----><select id="input-closing-date" class="custom-select">
                        <option value="24">24日</option>
                        <option value="25">25日</option>
                      </select>
                      <!---->
                    </div>
                  </div>
                </div>
              </div>
              <div class="text-center"><button type="button" class="btn btn-secondary rounded-pill">
                  キャンセル
                </button> <button type="button" class="btn mr-2 btn-color-active-import btn-secondary rounded-pill">
                  OK
                </button></div>
            </div>
            <!---->
          </div>
          <!---->
        </div>
      </div>
    </transition-stub>
    <transition-stub name="" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade">
      <!---->
    </transition-stub>
  </div>
</div>
`;

exports[`TEST COMPONENT LIST SHIFT Check click button Excel 2`] = `
<div class="col">
  <div class="list-shift">
    <div class="list-shift__header">
      <div class="row">
        <div class="col-sm-12 col-md-8 col-lg-8 col-xl-8 col-12">
          <div class="zone-left">
            <div class="zone-title"><span class="title-page">
                                LIST_SHIFT.TITLE_LIST_SHIFT_PRACTICAL_RECORD_TABLE
                            </span></div>
            <div class="zone-select-week-month"></div>
            <div class="zone-calendar-week"></div>
          </div>
        </div>
        <div class="col-sm-12 col-md-4 col-lg-4 col-xl-4 col-12">
          <!---->
          <!---->
          <div class="zone-right" style="display: none;">
            <div class="item-function btn-excel">
              <div class="show-icon"><i class="fas fa-file-excel"></i></div>
              <div class="show-text"><span>LIST_SHIFT.BUTTON_DOWNLOAD_EXCEL</span></div>
            </div>
          </div>
          <!---->
        </div>
      </div>
    </div>
    <div class="component-line-gray"></div>
    <div class="list-shift__control">
      <div class="row">
        <div class="col">
          <div class="text-left">
            <div role="group" class="button-text-left btn-group"><button type="button" class="btn btn-secondary control-button-group"><span>LIST_SHIFT.BUTTON_SHIFT_TABLE</span></button> <button type="button" class="btn btn-secondary control-button-group"><span>LIST_SHIFT.BUTTON_HIGHT_WAY_FEE</span></button> <button type="button" class="btn btn-secondary control-button-group"><span>LIST_SHIFT.BUTTON_TABLE_SALES</span></button> <button type="button" class="btn btn-secondary control-button-group"><span>LIST_SHIFT.BUTTON_PAYMENT</span></button></div>
          </div>
        </div>
        <div class="col">
          <div class="text-right"><button type="button" class="btn btn-edit btn-color-active btn-secondary rounded-pill" style="display: none;">
              LIST_SHIFT.BUTTON_EDIT
            </button></div>
          <!---->
        </div>
      </div>
    </div>
    <div class="list-shift__table">
      <div class="b-overlay-wrap position-relative">
        <div class="zone-table">
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
        <transition-stub name="" appear="true" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade" appearclass="" appearactiveclass="fade" appeartoclass="show">
          <!---->
        </transition-stub>
      </div>
    </div>
  </div>
  <div id="modal-closing-date___BV_modal_outer_" style="position: absolute; z-index: 1040;">
    <transition-stub enterclass="" leaveclass="" entertoclass="" leavetoclass="" enteractiveclass="" leaveactiveclass="">
      <div id="modal-closing-date" role="dialog" aria-hidden="true" aria-describedby="modal-closing-date___BV_modal_body_" class="modal fade" style="display: none;">
        <div class="modal-dialog modal-md">
          <!---->
          <div id="modal-closing-date___BV_modal_content_" tabindex="-1" class="modal-content">
            <!---->
            <div id="modal-closing-date___BV_modal_body_" class="modal-body modal-closing-date">
              <div class="text-center">
                <h5 class="font-weight-bolde">
                  締日を選択してください
                </h5>
              </div>
              <div class="body-item">
                <div class="row">
                  <div class="col-8" style="margin: auto;">
                    <div role="group" class="input-group">
                      <!----><select id="input-closing-date" class="custom-select">
                        <option value="24">24日</option>
                        <option value="25">25日</option>
                      </select>
                      <!---->
                    </div>
                  </div>
                </div>
              </div>
              <div class="text-center"><button type="button" class="btn btn-secondary rounded-pill">
                  キャンセル
                </button> <button type="button" class="btn mr-2 btn-color-active-import btn-secondary rounded-pill">
                  OK
                </button></div>
            </div>
            <!---->
          </div>
          <!---->
        </div>
      </div>
    </transition-stub>
    <transition-stub name="" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade">
      <!---->
    </transition-stub>
  </div>
</div>
`;
